### PR TITLE
feat: add typed-express-router package

### PR DESCRIPTION
**Pull request by bitgopatmcl**
Initial PR for a lighter weight integration of api-ts specifications with Express. The idea is to provide a relatively unopinionated base layer that provides type-checked requests and is flexible enough for other packages to build off of. This should improve familiarity and DX while simultaneously not reinventing the wheel on much of the functionality the existing router provides.

## Goals

- Define routes in Express that are associated with an api-ts `apiSpec`
- Augment the existing Express request with the decoded request object
- Augment the existing Express response with a type-checked `encode` function
- Allow customization of what to do on decode/encode errors, per-route if desired
- Allow action to be performed after an encoded response is sent, per-route if desired
- Allow routes to be defined with path that is different than the one specified in the
  `httpRoute` (e.g. for aliases)
- Follow the express router api as closely as possible otherwise

## Non-Goals

- Enforce that all routes listed in an `apiSpec` have an associated route handler
- Layer anything on top of the `express.RequestHandler[]` chain beyond the additional
  properties described above (projects and other libraries can do this)

### CODE REVIEW

**Comment by bitgopatmcl** [comment]
Should resolve #135 both with the after-response hook and by not messing with request handler chains. Also should address #189.

File: packages/typed-express-router/src/errors.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+) {
+  res.status(400).send('decode error IMPROVE THIS MESSAGE').end();
+}
+
+export function defaultOnEncodeError(
+  _err: unknown,
+  _req: express.Request,
+  res: express.Response,
+) {
+  res.status(500).send('Internal server error IMPROVE THIS MESSAGE').end();
```
Comment (ericcrosson-bitgo) [inline packages/typed-express-router/src/errors.ts:17]: Got a ticket for these action items? :grinning:

**Review by ericcrosson-bitgo — DISMISSED** [review]
The main file `index.ts` file looks pretty simple, true to the idea of this new package. Thanks for the high test coverage! Plenty of examples of use made it easy to review this code

Looks great to me, I won't merge after approval in case you wanted to manually add a git tag that semantic-release will pick up as an input in calculating the next package version. I _think_ if we don't add a tag it'll pick some beta form of 1.0.0 (but on the beta dist tag, [FAQ](https://semantic-release.gitbook.io/semantic-release/support/faq#can-i-set-the-initial-release-version-of-my-package-to-0.0.1)). I have no idea what happens when we drop out of beta in that case, whether we land on v1 or v2

File: packages/typed-express-router/src/errors.ts (hunk 1 above)
Comment (bitgopatmcl) [reply packages/typed-express-router/src/errors.ts:17]: Oops, can copy over the ones from `express-wrapper` for now but still will make one for improving those, particularly the decode error one.

**Review by ericcrosson-bitgo — APPROVED** [review]

**Pull request merged**